src/Equilibrium/EquilibriumTypes.jl

```diff
@@ -0,0 +1,339 @@
+
+A mutable struct holding parameters for the Solev'ev (SOL) plasma equilibrium model.
+
+## Fields:
+
+- `mr`: number of radial grid zones
+- `mz`: number of axial grid zones
+- `ma`: number of flux grid zones
+- `e`:  elongation
+- `a`: minor radius
+- `r0`: major radius
+- `q0`: safety factor at the o-point
+- `p0fac`: scale on-axis pressure (P-> P+P0*p0fac. beta changes. Phi,q constant)
+- `b0fac`: scale toroidal field at constant beta (s*Phi,s*f,s^2*P. bt changes. Shape,beta constant)
+- `f0fac`: scale toroidal field at constant pressure (s*f. beta,q changes. Phi,p,bp constant)
+"""
+
+@kwdef mutable struct SolevevConfig
+    mr::Int = 128      # number of radial grid zones
+    mz::Int = 128      # number of axial grid zones
+    ma::Int = 128      # number of flux grid zones
+    e::Float64 = 1.6       # elongation
+    a::Float64 = 0.33      # minor radius
+    r0::Float64 = 1.0      # major radius
+    q0::Float64 = 1.9      # safety factor at the o-point
+    p0fac::Float64 = 1       # scale on-axis pressure (P-> P+P0*p0fac. beta changes. Phi,q constant)
+    b0fac::Float64 = 1       # scale toroidal field at constant beta (s*Phi,s*f,s^2*P. bt changes. Shape,beta constant)
+    f0fac::Float64 = 1       # scale toroidal field at constant pressure (s*f. beta,q changes. Phi,p,bp constant)
+end
+
+"""
+Outer constructor for LarConfig that enables a toml file 
+    interface for specifying the configuration settings
+"""
+function SolevevConfig(path::String) # if we use @kwdef, it generates SolevevConfig() so it conflicts with this line.
+    raw = TOML.parsefile(path)
+    input_data = get(raw, "SOL_INPUT", Dict())
+    return SolevevConfig(; symbolize_keys(input_data)...)
+end
+
+
+"""
+    DirectRunInput(...)
+
+A container struct that bundles all necessary inputs for the `direct_run` function.
+It is created by the `_read_efit` function after parsing the raw equilibrium file
+and preparing the initial splines.
+
+## Fields:
+- `equil_input`: The original `EquilInput` object.
+- `sq_in`
+        # x value: psin
+        # Quantity 1: F = R*Bt  [m T]
+        # Quantity 2: mu0 * Pressure (non-negative) [nt^2 / m^2 * mu0 = T^2]
+        # Quantity 3: q-profile 
+        # Quantity 4: sqrt(psi_norm)
+- `psi_in`:
+        # x, y value: R, Z [m]
+        # z value : poloidal flux adjusted to be zero at the boundary [Weber/radian]
+            # 1. ψ(R,Z) = ψ_boundary - ψ(R,Z)
+            # 2. if ψ = ψ * sign(ψ(centerR,centerZ))
+- `rmin`: Minimum R-coordinate of the computational grid [m].
+- `rmax`: Maximum R-coordinate of the computational grid [m].
+- `zmin`: Minimum Z-coordinate of the computational grid [m].
+- `zmax`: Maximum Z-coordinate of the computational grid [m].
+- `psio`: The total flux difference `abs(ψ_axis - ψ_boundary)` [Weber / radian].
+"""
+mutable struct DirectRunInput
+    config::EquilConfig
+    sq_in::Any       # 1D profile spline (CubicSplineType)
+    psi_in::Any      # 2D flux spline (BicubicSplineType)
```

Comment:
`Any` is always bad. Give these concrete type or use [parametric types](https://docs.julialang.org/en/v1/manual/types/#Parametric-Types)